test/matrix/csr_kernels2.cpp

```diff
@@ -191,6 +202,68 @@ class Csr : public CommonTestFixture {
         dmtx->copy_from(mtx);
     }
 
+    void test_mixed_spmv()
```

Comment:
all of these tests could also be run in `test/matrix.cpp` rather than duplicating the work across all mixed-precision capable matrix formats.